1Password/shell-plugins
# New plugin: AWS CDK Toolkit

## Summary

Add support for a new plugin for the AWS CDK Toolkit.

The AWS Cloud Development Kit (AWS CDK) lets you define your cloud infrastructure as code in one of its supported programming languages. You can use it to create, manage, deploy, etc. AWS CDK apps.

## Thought Process

The AWS CDK Toolkit uses the same credentials as the regular AWS CLI, and the same profile workflow. We can reuse the importers and provisioners that have already been created for the AWS shell plugin.

## How To Test

1. Install the AWS CDK Toolkit following [these instructions](https://docs.aws.amazon.com/cdk/v2/guide/cli.html)
2. Initialize the AWS CDK plugin by running `op plugin init cdk`
3. Create an AWS CDK app following [these instructions](https://docs.aws.amazon.com/cdk/v2/guide/hello_world.html)
4. Run a `cdk` command from the CDK app directory that requires authentication, e.g. `cdk deploy`. Try authenticating in the following ways. The command should be authenticated and run as expected:
  a. Use an existing `AWS Access Key` created for the regular AWS shell plugin
  b. Import credentials from the `~/.aws/credentials` file
  c. Import credentials manually as prompted
5. Run `make aws/validate`. All validation checks should pass and there should be a `# Executable: AWS CDK Toolkit` section
